src/commands/CreateDataSourceCommand.ts

```diff
@@ -21,17 +21,17 @@ import * as vscode from 'vscode';
 import { DataSourcesTreeNode } from '../model/tree/DataSourcesTreeNode';
 import { DataVirtConfig, DataSourceConfig, Property } from '../model/DataVirtModel';
 
-export async function createDataSourceCommand(dsTreeNode: DataSourcesTreeNode) {
+export async function createDataSourceCommand(dsNode: DataSourcesTreeNode) {
 	const dsName: string = await vscode.window.showInputBox( { validateInput: (name: string) => {
 		let msg: string = utils.validateName(name);
 		if (!msg) {
-			if (utils.getDataSourceByName(dsTreeNode.getProject().dvConfig, name)) {
+			if (utils.getDataSourceByName(dsNode.getProject().dvConfig, name)) {
 				msg = `There is already a datasource with the name ${name}.`;
 			}
 		}
 		return msg;
 	}, placeHolder: 'Enter the name of the new datasource'});
-	if (!dsName) {
+	if (dsName === undefined) {
```

Comment:
Are we ever hitting this? What's the condition if we do? Might want to check for an empty string as well?

Reply:
Yes, you hit that whenever you hit ESC inside an input dialog.